lib/optimizely/notification_center.rb

```diff
@@ -0,0 +1,154 @@
+#
+#    Copyright 2017, Optimizely and contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+module Optimizely
+  class NotificationCenter
+    
+    attr_reader :notifications
+    
+    NOTIFICATION_TYPES = {
+      DECISION: 'DECISION: experiment, user_id, attributes, variation, event',
+      TRACK: 'TRACK: event_key, user_id, attributes, event_tags, event',
+      FEATURE_ROLLOUT: 'FEATURE_ROLLOUT:feature_key, user_id, attributes, audiences',
+      FEATURE_EXPERIMENT: 'FEATURE_EXPERIMENT:feature_key, user_id, attributes, experiment, variation'
+    }.freeze
+
+    def initialize(logger, error_handler)
+      @notification_id = 1
+      @notifications = {}
+      NOTIFICATION_TYPES.values.each { |value| @notifications[value] = [] }
+      @logger = logger
+      @error_handler = error_handler
+    end
+
+    def add_notification_listener(notification_type, notification_callback)
+      # Adds notification callback to the notification center
+
+      # Args:
+      #  notification_type: DECISION
```

Comment:
DECISION is not the only notification type. Write for example one of the constants in NOTIFICATION_TYPES.